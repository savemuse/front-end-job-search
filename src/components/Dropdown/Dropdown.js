import React, { Component } from 'react';
import SelectField from 'material-ui/SelectField';
import MenuItem from 'material-ui/MenuItem';

class Dropdown extends Component {
  shouldComponentUpdate(nextProps) {
    return nextProps.filter !== this.props.filter || 
      nextProps.menuList.length !== this.props.menuList.length;
  }
  
  handleChange = (event, index, value) => {
    this.props.handleFilterChange(value);
  };

  render() {
    const { menuList, filter } = this.props;
    
    return (
      <SelectField
        hintText="Filter labels"
        floatingLabelText="Filter labels"
        value={filter}
        onChange={this.handleChange}
        maxHeight={200}
      >
        {menuList && menuList.map((item, idx) => (
          <MenuItem
            key={idx}
            value={item}
            primaryText={item} 
          />
        ))}
      </SelectField>
    );
  }
}

export default Dropdown;