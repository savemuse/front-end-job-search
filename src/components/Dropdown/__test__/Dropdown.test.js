import React from 'react';
import PropTypes from 'prop-types';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import MenuItem from 'material-ui/MenuItem';
import getMuiTheme from 'material-ui/styles/getMuiTheme';
import Dropdown from '../Dropdown';
const muiTheme = getMuiTheme();

describe('<Dropdown />', function () {
  let dropdown;
  beforeEach(() => {
    jest.resetModules();
    this.params = {
      menuList: [],
      filter: undefined,
      handleFilterChange: jest.fn()
    };

    this.makeSubject = () => {
      const { 
        menuList,
        filter,
        handleFilterChange
      } = this.params;

      dropdown = shallow(
        <Dropdown
          menuList={menuList}
          filter={filter}
          handleFilterChange={handleFilterChange}
        />, 
        {
          context: {muiTheme},
          childContextTypes: {muiTheme: PropTypes.object}
        }
      );
      return dropdown;
    };
  });

  describe('when it has complete data', () => {
    beforeEach(() => {
      this.params.menuList = [
        'GIT',
        'unit test',
        'Full Stack',
        '[F] React',
        '[F] Vue.js',
        '[F] D3.js',
        '[B] NodeJS',
        '[B] GraphQL',
        '[B] ROR',
        '年薪百萬',
      ]

      this.subject = this.makeSubject();
    });

    it('should be same as snapshot', () => {
      expect(toJson(dropdown)).toMatchSnapshot();
    });

    it('should be equal `GIT` for first option', () => {
      expect(dropdown.find(MenuItem).first().props().value).toEqual('GIT');
      expect(dropdown.find(MenuItem).at(0).props().primaryText).toEqual('GIT');
    });

    it('should be equal `unit test` for second option', () => {
      expect(dropdown.find(MenuItem).at(1).props().primaryText).toEqual('unit test');
    });
  });
});