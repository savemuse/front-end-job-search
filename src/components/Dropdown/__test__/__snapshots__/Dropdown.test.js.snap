// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Dropdown /> when it has complete data should be same as snapshot 1`] = `
<SelectField
  autoWidth={false}
  disabled={false}
  floatingLabelText="Filter labels"
  fullWidth={false}
  hintText="Filter labels"
  maxHeight={200}
  multiple={false}
  onChange={[Function]}
>
  <MenuItem
    anchorOrigin={
      Object {
        "horizontal": "right",
        "vertical": "top",
      }
    }
    checked={false}
    desktop={false}
    disabled={false}
    focusState="none"
    insetChildren={false}
    primaryText="GIT"
    targetOrigin={
      Object {
        "horizontal": "left",
        "vertical": "top",
      }
    }
    value="GIT"
  />
  <MenuItem
    anchorOrigin={
      Object {
        "horizontal": "right",
        "vertical": "top",
      }
    }
    checked={false}
    desktop={false}
    disabled={false}
    focusState="none"
    insetChildren={false}
    primaryText="unit test"
    targetOrigin={
      Object {
        "horizontal": "left",
        "vertical": "top",
      }
    }
    value="unit test"
  />
  <MenuItem
    anchorOrigin={
      Object {
        "horizontal": "right",
        "vertical": "top",
      }
    }
    checked={false}
    desktop={false}
    disabled={false}
    focusState="none"
    insetChildren={false}
    primaryText="Full Stack"
    targetOrigin={
      Object {
        "horizontal": "left",
        "vertical": "top",
      }
    }
    value="Full Stack"
  />
  <MenuItem
    anchorOrigin={
      Object {
        "horizontal": "right",
        "vertical": "top",
      }
    }
    checked={false}
    desktop={false}
    disabled={false}
    focusState="none"
    insetChildren={false}
    primaryText="[F] React"
    targetOrigin={
      Object {
        "horizontal": "left",
        "vertical": "top",
      }
    }
    value="[F] React"
  />
  <MenuItem
    anchorOrigin={
      Object {
        "horizontal": "right",
        "vertical": "top",
      }
    }
    checked={false}
    desktop={false}
    disabled={false}
    focusState="none"
    insetChildren={false}
    primaryText="[F] Vue.js"
    targetOrigin={
      Object {
        "horizontal": "left",
        "vertical": "top",
      }
    }
    value="[F] Vue.js"
  />
  <MenuItem
    anchorOrigin={
      Object {
        "horizontal": "right",
        "vertical": "top",
      }
    }
    checked={false}
    desktop={false}
    disabled={false}
    focusState="none"
    insetChildren={false}
    primaryText="[F] D3.js"
    targetOrigin={
      Object {
        "horizontal": "left",
        "vertical": "top",
      }
    }
    value="[F] D3.js"
  />
  <MenuItem
    anchorOrigin={
      Object {
        "horizontal": "right",
        "vertical": "top",
      }
    }
    checked={false}
    desktop={false}
    disabled={false}
    focusState="none"
    insetChildren={false}
    primaryText="[B] NodeJS"
    targetOrigin={
      Object {
        "horizontal": "left",
        "vertical": "top",
      }
    }
    value="[B] NodeJS"
  />
  <MenuItem
    anchorOrigin={
      Object {
        "horizontal": "right",
        "vertical": "top",
      }
    }
    checked={false}
    desktop={false}
    disabled={false}
    focusState="none"
    insetChildren={false}
    primaryText="[B] GraphQL"
    targetOrigin={
      Object {
        "horizontal": "left",
        "vertical": "top",
      }
    }
    value="[B] GraphQL"
  />
  <MenuItem
    anchorOrigin={
      Object {
        "horizontal": "right",
        "vertical": "top",
      }
    }
    checked={false}
    desktop={false}
    disabled={false}
    focusState="none"
    insetChildren={false}
    primaryText="[B] ROR"
    targetOrigin={
      Object {
        "horizontal": "left",
        "vertical": "top",
      }
    }
    value="[B] ROR"
  />
  <MenuItem
    anchorOrigin={
      Object {
        "horizontal": "right",
        "vertical": "top",
      }
    }
    checked={false}
    desktop={false}
    disabled={false}
    focusState="none"
    insetChildren={false}
    primaryText="年薪百萬"
    targetOrigin={
      Object {
        "horizontal": "left",
        "vertical": "top",
      }
    }
    value="年薪百萬"
  />
</SelectField>
`;
